h2o-algos/src/main/java/hex/glrm/GLRM.java

```diff
@@ -1131,13 +1186,29 @@ private Chunk chk_xnew(Chunk[] chks, int c) {
 
           // Gradient wrt x_i is matrix product \grad L_{i,j}(x_i * Y_j, A_{i,j}) * Y_j'
           double[] weight = _lossFunc[j].mlgrad(xy, (int) a[j]);
```

Comment:
Pasha:

Like this suggestion.  However, this will involve changes across several classes/methods.  I want to postpone this to PUBDEV-3560: GLRM Efficiency improvement: avoid re-allocating arrays on every iteration.

Wendy